R/plot.R

```diff
@@ -0,0 +1,127 @@
+#' Plot method for `<epidist>` class
+#'
+#' @description Plot an `<epidist>` object by displaying the either the
+#' probability mass function (PMF), (in the case of discrete distributions)
+#' or probability density function (PDF) (in the case of continuous
+#' distributions), or the cumulative distribution function (CDF).
+#'
+#' @details
+#' By default if the `xlim` argument is not specified the distribution is
+#' plotted between day 0 and the 99th quantile of the distribution.
+#' Alternatively, a `numeric` vector of length 2 with the
+#' first and last day to plot on the x-axis can be supplied to `xlim`
+#' (through [...]).
+#'
+#'
+#' @param x An `<epidist>` object.
+#' @param cumulative A boolean `logical`, default is `FALSE`.
+#' `cumulative = TRUE` plots the cumulative distribution function (CDF).
+#' @inheritParams base::print
+#'
+#' @author Joshua W. Lambert
+#' @export
+#'
+#' @examples
+#' # plot continuous epidist
+#' edist <- epidist(
+#'   disease = "ebola",
+#'   epi_dist = "incubation_period",
+#'   prob_distribution = "gamma",
+#'   prob_distribution_params = c(shape = 2, scale = 1)
+#' )
+#' plot(edist)
+#'
+#' # plot different day range (x-axis)
+#' plot(edist, xlim = c(0, 10))
+#'
+#' # plot CDF
+#' plot(edist, cumulative = TRUE)
+#'
+#' # plot discrete epidist
+#' edist <- discretise(edist)
+#' plot(edist)
+plot.epidist <- function(x,
+                         cumulative = FALSE,
+                         ...) {
+  # check input
+  validate_epidist(x)
+  checkmate::assert_logical(cumulative, any.missing = FALSE, len = 1)
+
+  # capture dots
+  dots <- list(...)
+
+  oldpar <- graphics::par(no.readonly = TRUE)
+  on.exit(graphics::par(oldpar))
```

Comment:
As far as I can tell, this is no longer necessary

Reply:
I'm happy to remove these lines. 

Has there been a recent change that has led to these not being needed? If so could you share the link?